src/main/java/fi/aalto/cs/apluscourses/utils/FileDateFormatter.java

```diff
@@ -0,0 +1,86 @@
+package fi.aalto.cs.apluscourses.utils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.Clock;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+public class FileDateFormatter {
+  // the order of time units to check, ordered from most to least precise
+  private static final List<ChronoUnit> TimeUnits = Arrays.asList(
+          ChronoUnit.SECONDS,
+          ChronoUnit.MINUTES,
+          ChronoUnit.HOURS,
+          ChronoUnit.DAYS,
+          ChronoUnit.MONTHS,
+          ChronoUnit.YEARS
+  );
+
+  private static String formatWithTimeUnit(ZonedDateTime fileTime,
+                                           ZonedDateTime currentTime,
+                                           ChronoUnit timeUnit) {
+    long timeUnitCount = timeUnit.between(fileTime, currentTime);
+    String timeUnitText = timeUnit.toString().toLowerCase(Locale.ROOT);
+    if (timeUnitCount == 1) {
+      // remove the final pluralizing "s" from the time unit name
+      timeUnitText = timeUnitText.substring(0, timeUnitText.length() - 1);
+    }
+
+    return timeUnitCount + " " + timeUnitText + " ago";
+  }
+
+  private static String formatWithLargestTimeUnit(ZonedDateTime fileTime, Clock currentTimeClock) {
+    ZonedDateTime currentTime = ZonedDateTime.now(currentTimeClock);
+
+    if (ChronoUnit.SECONDS.between(fileTime, currentTime) <= 0) {
+      return "just now";
+    }
+
+    for (int i = 1; i < TimeUnits.size(); i++) {
+      if (TimeUnits.get(i).between(fileTime, currentTime) == 0) {
+        return formatWithTimeUnit(fileTime, currentTime, TimeUnits.get(i - 1));
+      }
+    }
+
+    return formatWithTimeUnit(fileTime, currentTime, TimeUnits.get(TimeUnits.size() - 1));
+  }
+
+  /**
+   * Same as {@link FileDateFormatter#getFileModificationTime(Path, Clock)} with the clock
+   * being always set to the system's current time.
+   *
+   * @see FileDateFormatter#getFileModificationTime(Path, Clock)
+   */
+
+  public static String getFileModificationTime(Path filePath) throws IOException {
+    return getFileModificationTime(filePath, Clock.system(ZoneOffset.UTC));
```

Comment:
This isn't a comment on the code per se, but just making sure I understand this correctly: is it correct that it wouldn't matter what zone offset you use here, since the file time is also converted to the time zone of the `currentTimeClock` parameter?

Reply:
Correct, no matter which time zone you use here, it will work the same. It's just that time zone is a mandatory parameter, so I went with the most "standard" one - UTC. Speaking of which, I just found out that there's already a wrapper function which does the same: `Clock.systemUTC()`

Reply:
Sounds good :+1: Might even add a comment like `// any time zone would work here`, my initial reaction to seeing UTC was "surely this can't be correct"